Lib/_pydatetime.py

```diff
@@ -161,14 +161,23 @@ def _build_struct_time(y, m, d, hh, mm, ss, dstflag):
     dnum = _days_before_month(y, m) + d
     return _time.struct_time((y, m, d, hh, mm, ss, wday, dnum, dstflag))
 
-def _format_time(hh, mm, ss, us, timespec='auto'):
-    specs = {
-        'hours': '{:02d}',
-        'minutes': '{:02d}:{:02d}',
-        'seconds': '{:02d}:{:02d}:{:02d}',
-        'milliseconds': '{:02d}:{:02d}:{:02d}.{:03d}',
-        'microseconds': '{:02d}:{:02d}:{:02d}.{:06d}'
-    }
+def _format_time(hh, mm, ss, us, timespec='auto', basic=False):
+    if basic:
+        specs = {
+            'hours': '{:02d}',
+            'minutes': '{:02d}{:02d}',
+            'seconds': '{:02d}{:02d}{:02d}',
+            'milliseconds': '{:02d}{:02d}{:02d}.{:03d}',
+            'microseconds': '{:02d}{:02d}{:02d}.{:06d}'
+        }
+    else:
+        specs = {
+            'hours': '{:02d}',
+            'minutes': '{:02d}:{:02d}',
+            'seconds': '{:02d}:{:02d}:{:02d}',
+            'milliseconds': '{:02d}:{:02d}:{:02d}.{:03d}',
+            'microseconds': '{:02d}:{:02d}:{:02d}.{:06d}'
+        }
```

Comment:
```suggestion
    specs = {
        'hours': '{:02d}',
        'minutes': '{:02d}{:02d}',
        'seconds': '{:02d}{:02d}{:02d}',
        'milliseconds': '{:02d}{:02d}{:02d}.{:03d}',
        'microseconds': '{:02d}{:02d}{:02d}.{:06d}'
    }
    if not basic:
         specs = {k: v.replace("}{", "}:{") for k, v in specs.items()}

```

Reply:
Doing the opposite will lower the diff here by a lot, apologies my suggestion is the opposite by accident!

Reply:
I prefer keeping explicit dicts. It's clearer to the reader.

Reply:
```suggestion
    specs = {
        'hours': '{:02d}',
        'minutes': '{:02d}:{:02d}',
        'seconds': '{:02d}:{:02d}:{:02d}',
        'milliseconds': '{:02d}:{:02d}:{:02d}.{:03d}',
        'microseconds': '{:02d}:{:02d}:{:02d}.{:06d}'
    }
    # Remove colons for the basic format
    if basic:
         specs = {k: v.replace("}:{", "}{") for k, v in specs.items()}
```

IMO a comment is clear enough, these are also tests, not source, so they don't have to be perfect ;-)

Reply:
I still prefer my form. And those are in `Lib/_pydatetime.py` so they are still in `Lib` and not in `Lib/test`. If a codeowner thinks differently, I won't mind changing my mind as they're the principal maintainers, but I think it's better to have symmetric branches. We're doing useless iterations IMO.